src/main/kotlin/solve/importer/ProjectParser.kt

```diff
@@ -46,23 +45,23 @@ object ProjectParser {
             landmark.projectLayer
         }.toSet())
         val diffString = diff.map { layer ->
-            layer.name + "_" + layer.kind
+            layer.name
+        }
+
+        fun addNoSomeAlgorithmErrorMessage(toBe: String, ending: String) {
+            frame.errorMessage.add(
+                "There $toBe no ${
+                    diffString.toMutableList().toStringWithoutBrackets()
+                } algorithm$ending for image ${frame.frame.timestamp}"
+            )
         }
 
         if (diff.isNotEmpty()) {
             frame.isImageErrored = true
             if (diff.count() == 1) {
-                frame.errorMessage.add(
-                    "There is no ${
-                        diffString.toMutableList().toStringWithoutBrackets()
-                    } algorithm for image ${frame.frame.timestamp}"
-                )
+                addNoSomeAlgorithmErrorMessage("is", "")
             } else {
-                frame.errorMessage.add(
-                    "There are no ${
-                        diffString.toMutableList().toStringWithoutBrackets()
-                    } algorithms for image ${frame.frame.timestamp}"
-                )
+                addNoSomeAlgorithmErrorMessage("are", "s")
```

Comment:
decide inside special func by logic, not hardcode here))

Reply:
so, to put this `if` into a function `addNoSomeAlgorithmErrorMessage`?

Reply:
I tried to avoid code duplication. And if you take out `if`, then it will be inside the function

Reply:
yes, you can put if in such way that only different code is in it, while the same one is not duplicated)